### PR TITLE
Merge dav1d 0.1.0 x86 assembly

### PR DESCRIPTION
**Pull request by barrbrain**
Advances #1750.

Difference between upstream commit list and this PR:
```diff
+Integrate inv_txfm_add_..._4x4_ssse3 from itx_ssse3.asm
+Integrate ipred_h_ssse3 from ipred_ssse3.asm
+asm/x86: Use list syntax in macros
+Integrate avg_ssse3 from mc_ssse3.asm
+Add rav1e_dr_intra_derivative table for asm
-cdef: simplify sec_taps
+Add rav1e_mc_warp_filter table for asm
-Add a 4x4 cdef_filter AVX2 implementation
-Add 8x8 cdef_filter AVX2 implementation
-Add AVX2 SIMD implementation for cdef_dir
+Add rav1e_sgr_x_by_xplus1 table for asm
+Add rav1e_filter_intra_taps table for asm
-x86: Add chroma-from-luma intra prediction AVX2 asm
-x86: Add smooth intra prediction AVX2 asm
-x86: Add paeth intra prediction AVX2 asm
-x86: Add dc/h/v intra prediction AVX2 asm
-x86: Enable ITX AVX2 asm on 64-bit Windows
-x86: Fix out-of-bounds write in ITX asm
-x86-64: Inverse transforms AVX2
-x86: Enable MC AVX2 asm on 64-bit Windows
-x86: MC AVX2
-x86 asm infrastructure
+Move mc_subpel_filters table from x86 asm to shared Rust
```

The first 2 MC commits and the "x86 asm infrastructure" commit were already squashed in #639.
The ITX commits were squashed in #864, along with other commits from dav1d-0.2.0.
Our CDEF assembly is divergent from dav1d, and appears to be a subset of code in 0.2.0.

### CODE REVIEW

File: src/x86/cpuid.asm [hunk 1]
```diff
@@ -0,0 +1,55 @@
+; Copyright © 2018, VideoLAN and dav1d authors
```
Comment (lu-zero) [inline src/x86/cpuid.asm]: It does not seem used from a first look and seems not needed.

Comment (barrbrain) [reply src/x86/cpuid.asm]: I will drop the first commit on the next push. This code is obviated by the Rust cpu-feature implementation.

**Comment by barrbrain** [comment]
New symbols in existing files:
- `rav1e_blend_avx2`
- `rav1e_blend_h_avx2`
- `rav1e_blend_v_avx2`
- `rav1e_emu_edge_avx2`
- `rav1e_ipred_cfl_ac_420_avx2`
- `rav1e_ipred_filter_avx2`
- `rav1e_ipred_z1_avx2`
- `rav1e_pal_pred_avx2`
- `rav1e_warp_affine_8x8_avx2`
- `rav1e_warp_affine_8x8t_avx2`

New symbols in new files:
- `rav1e_avg_ssse3`
- `rav1e_blend_h_ssse3`
- `rav1e_blend_ssse3`
- `rav1e_blend_v_ssse3`
- `rav1e_iadst_4x4_internal_ssse3`
- `rav1e_idct_4x4_internal_ssse3`
- `rav1e_iflipadst_4x4_internal_ssse3`
- `rav1e_iidentity_4x4_internal_ssse3`
- `rav1e_inv_txfm_add_adst_adst_4x4_ssse3`
- `rav1e_inv_txfm_add_adst_dct_4x4_ssse3`
- `rav1e_inv_txfm_add_adst_flipadst_4x4_ssse3`
- `rav1e_inv_txfm_add_adst_identity_4x4_ssse3`
- `rav1e_inv_txfm_add_dct_adst_4x4_ssse3`
- `rav1e_inv_txfm_add_dct_dct_4x4_ssse3`
- `rav1e_inv_txfm_add_dct_flipadst_4x4_ssse3`
- `rav1e_inv_txfm_add_dct_identity_4x4_ssse3`
- `rav1e_inv_txfm_add_flipadst_adst_4x4_ssse3`
- `rav1e_inv_txfm_add_flipadst_dct_4x4_ssse3`
- `rav1e_inv_txfm_add_flipadst_flipadst_4x4_ssse3`
- `rav1e_inv_txfm_add_flipadst_identity_4x4_ssse3`
- `rav1e_inv_txfm_add_identity_adst_4x4_ssse3`
- `rav1e_inv_txfm_add_identity_dct_4x4_ssse3`
- `rav1e_inv_txfm_add_identity_flipadst_4x4_ssse3`
- `rav1e_inv_txfm_add_identity_identity_4x4_ssse3`
- `rav1e_inv_txfm_add_wht_wht_4x4_ssse3`
- `rav1e_ipred_h_ssse3`
- `rav1e_lpf_h_sb_uv_avx2`
- `rav1e_lpf_h_sb_y_avx2`
- `rav1e_lpf_v_sb_uv_avx2`
- `rav1e_lpf_v_sb_y_avx2`
- `rav1e_mask_ssse3`
- `rav1e_sgr_box3_h_avx2`
- `rav1e_sgr_box3_v_avx2`
- `rav1e_sgr_box5_h_avx2`
- `rav1e_sgr_box5_v_avx2`
- `rav1e_sgr_calc_ab1_avx2`
- `rav1e_sgr_calc_ab2_avx2`
- `rav1e_sgr_finish_filter1_avx2`
- `rav1e_sgr_finish_filter2_avx2`
- `rav1e_sgr_weighted1_avx2`
- `rav1e_sgr_weighted2_avx2`
- `rav1e_w_avg_ssse3`
- `rav1e_w_mask_420_ssse3`
- `rav1e_wiener_filter_h_avx2`
- `rav1e_wiener_filter_v_avx2`

**Comment by barrbrain** [comment]
New SSSE3 implementations with existing AVX2 implementations:
- `rav1e_avg_ssse3`
- `rav1e_iadst_4x4_internal_ssse3`
- `rav1e_idct_4x4_internal_ssse3`
- `rav1e_iflipadst_4x4_internal_ssse3`
- `rav1e_iidentity_4x4_internal_ssse3`
- `rav1e_inv_txfm_add_adst_adst_4x4_ssse3`
- `rav1e_inv_txfm_add_adst_dct_4x4_ssse3`
- `rav1e_inv_txfm_add_adst_flipadst_4x4_ssse3`
- `rav1e_inv_txfm_add_adst_identity_4x4_ssse3`
- `rav1e_inv_txfm_add_dct_adst_4x4_ssse3`
- `rav1e_inv_txfm_add_dct_dct_4x4_ssse3`
- `rav1e_inv_txfm_add_dct_flipadst_4x4_ssse3`
- `rav1e_inv_txfm_add_dct_identity_4x4_ssse3`
- `rav1e_inv_txfm_add_flipadst_adst_4x4_ssse3`
- `rav1e_inv_txfm_add_flipadst_dct_4x4_ssse3`
- `rav1e_inv_txfm_add_flipadst_flipadst_4x4_ssse3`
- `rav1e_inv_txfm_add_flipadst_identity_4x4_ssse3`
- `rav1e_inv_txfm_add_identity_adst_4x4_ssse3`
- `rav1e_inv_txfm_add_identity_dct_4x4_ssse3`
- `rav1e_inv_txfm_add_identity_flipadst_4x4_ssse3`
- `rav1e_inv_txfm_add_identity_identity_4x4_ssse3`
- `rav1e_inv_txfm_add_wht_wht_4x4_ssse3`
- `rav1e_ipred_h_ssse3`
- `rav1e_mask_ssse3`
- `rav1e_w_avg_ssse3`
- `rav1e_w_mask_420_ssse3`

**Comment by tdaede** [comment]
> Our CDEF assembly is divergent from dav1d, and appears to be a subset of code in 0.2.0.

This is because dav1d CDEF clobbers the input buffer (unless there are other reasons I don't know about?)

**Comment by barrbrain** [comment]
Observed 6.1%~7.8% speed up on [AWCY at speed 2 with `RAV1E_CPU_TARGET=ssse3`](https://beta.arewecompressedyet.com/?job=master-s2-cd1c5a69-ssse3%402019-10-16T07%3A53%3A19.802Z&job=merge-dav1d-0.1.0-s2-ssse3%402019-10-16T07%3A53%3A41.976Z) with no change in encoded output.

**Comment by lu-zero** [comment]
Sounds fairly good

**Comment by barrbrain** [comment]
Waiting on #1768 to be resolved and merged. Then this can be rebased, tables moved to a common module and the `sgr_x_by_x` table added for completeness.

**Pull request merged**